src/bson.rs

```diff
@@ -373,24 +373,24 @@ impl Bson {
     /// Get the `ElementType` of this value.
     pub fn element_type(&self) -> ElementType {
         match *self {
-            Bson::FloatingPoint(..) => ElementType::FloatingPoint,
-            Bson::String(..) => ElementType::Utf8String,
+            Bson::Double(..) => ElementType::Double,
+            Bson::String(..) => ElementType::String,
             Bson::Array(..) => ElementType::Array,
             Bson::Document(..) => ElementType::EmbeddedDocument,
-            Bson::Boolean(..) => ElementType::Boolean,
-            Bson::Null => ElementType::NullValue,
-            Bson::Regex(..) => ElementType::RegularExpression,
+            Bson::Bool(..) => ElementType::Bool,
+            Bson::Null => ElementType::Null,
+            Bson::Regex(..) => ElementType::Regex,
```

Comment:
mild preference for naming the actual type `RegularExpression`, no need to conserve on bytes

Reply:
Same rationale here again. It's less about conserving on bytes and more about consistency with the existing ecosystem. Both the crate and the type for regular expressions in Rust are `regex`.